DOCSP-27770 - Remove Placeholders (#74)
mongodb/docs-csharp: source/quick-start.txt

.. include:: /includes/quick-start/overview.rst

Create a MongoDB Cluster
------------------------

Set Up a Free Tier Cluster in Atlas
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Complete Chapter 1 of the

> :guides:`MongoDB Guide </atlas/account>` to set up a new

Author: I didn't know whether there's a standard way to refer to this guide (guides?). In this case, we want the user to complete just Chapter 1.

.. include:: /includes/quick-start/overview.rst

> Create a MongoDB Cluster

Author: Moved this section first so it goes from Create Cluster > Create Project > Code Project, rather than Create Project > Create Cluster > Code Project.

------------------------

> Set Up a Free Tier Cluster in Atlas

Author: Moved a lot of this content directly into this page instead of an `includes` file, since it's not used anywhere else.

After setting up your .NET project dependencies, create a MongoDB cluster
where you can store and manage your data. Complete the
:atlas:`Get Started with Atlas </getting-started>` guide to set up a new
Atlas account and free tier MongoDB cluster, load datasets, and
interact with the data.

After completing the steps in the Atlas guide, you have a new MongoDB
cluster deployed in Atlas, a new database user, and
:atlas:`sample datasets loaded </sample-data/>` into your cluster.

Find Your Connection String
~~~~~~~~~~~~~~~~~~~~~~~~~~~

To connect to the Atlas cluster you created in the previous step, you need the 
*connection string* for your cluster and user. A connection string contains the 
information the {+driver-short+} needs to connect 
to a MongoDB cluster, including the hostname or IP address and port of the
cluster, authentication mechanism, user credentials (when applicable), and
other connection options.

To retrieve the connection string for your Atlas cluster and user, complete the 
:guides:`Get Connection String </atlas/connection-string>`
guide. Once you have finished the guide, you will have a connection string similar to the 
following:

.. code-block:: bash

>    "mongodb+srv://username:password@cluster0.abc.mongodb.net/?retryWrites=true&w=majority"

Reviewer: Suggestion: Use brackets here to make the placeholders stand out more 

```suggestion
   "mongodb+srv://<username>:<password>@cluster0.abc.mongodb.net/?retryWrites=true&w=majority"
```

> Save your Atlas connection string to an environment variable named ``MONGODB_URI``. 

Author: Moved this step to immediately after they retrieve the string, rather than having them copy it down somewhere else first.

Reviewer: I think might be confusing here for some users since they haven't created the app or the project directory yet. If I were starting off from scratch and reading this I wouldn't know if this should be in the same location as the app or somewhere else entirely.
 
I also wonder if all users will know what exactly we mean by saying to "save to an environment variable", but that might be a separate discussion.

Author: Yeah, the "environment variable" language was already here, but we can talk about whether to clarify it. 

I don't think it makes a difference where they run this command, since it's an environment-wide variable. What if we explain the steps a little more?

.. include:: /includes/quick-start/overview.rst

Create a MongoDB Cluster
------------------------

Set Up a Free Tier Cluster in Atlas
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Complete the

> :atlas:`Get Started with Atlas </getting-started>` guide to set up a new

Reviewer 2: Issue:

This linked guide actually currently shows everything that this guide shows, including running CRUD operations. Therefore, if you complete that guide, there's no reason to continue with this one.

Suggestion:
You might consider linking to this one instead for now since it ends with "Get Connection String": https://www.mongodb.com/docs/guides/atlas/account/

> Atlas account and free tier MongoDB cluster, load datasets, and

Reviewer 2: ```suggestion
Atlas account and free tier MongoDB cluster, load sample datasets, and
```

interact with the data.

After completing the steps in the Atlas guide, you have a new MongoDB
cluster deployed in Atlas, a new database user, and
:atlas:`sample datasets loaded </sample-data/>` into your cluster.

Find Your Connection String
~~~~~~~~~~~~~~~~~~~~~~~~~~~

To connect to the Atlas cluster you created in the previous step, you need the 
*connection string* for your cluster and user. A connection string contains the 
information the {+driver-short+} needs to connect 
to a MongoDB cluster, including the hostname or IP address and port of the
cluster, authentication mechanism, user credentials (when applicable), and
other connection options.

To retrieve the connection string for your Atlas cluster and user, complete the 
:guides:`Get Connection String </atlas/connection-string>`
guide. Once you have finished the guide, you will have a connection string similar to the 
following:

.. code-block:: bash

   "mongodb+srv://<username>:<password>@cluster0.abc.mongodb.net/?retryWrites=true&w=majority"

> Run the following code at the command prompt to save your Atlas connection string to an 

Reviewer 2: Suggestion:

What do you think about creating a new section for this part, perhaps entitled "Set Your Connection String Environment Variable"? That helps logically separate it from "Find Your Connection String".

If a user reads https://www.mongodb.com/docs/guides/atlas/account/ (suggested link in an earlier comment) to completion, they should be able to start with their "connection string in [their] copy buffer".

   dotnet add package MongoDB.Driver

Create a MongoDB Cluster
------------------------

.. include:: /includes/quick-start/atlas-setup.rst

Query Your MongoDB Cluster from Your Application
------------------------------------------------

First, save your Atlas connection string to an environment variable named ``MONGODB_URI``. This prevents potential security issues caused by including your credentials in the source code.
In this step, you'll use the {+driver-short+} 
to connect to your MongoDB cluster and run a query on the sample data.

.. code-block:: bash

   export MONGODB_URI='<your atlas connection string>'

.. important::

>    Make sure to replace the "<username>" and "<password>" sections of the connection string with the username and password of your Atlas user.

Reviewer: Suggestion: Could we add a version of this admonition to where we tell the user to create their environment variable? I think it's still good to point this out in case some users are just copy/pasting and don't realize the placeholders are there.

   dotnet add package MongoDB.Driver

Query Your MongoDB Cluster from Your Application
------------------------------------------------

In this step, you'll use the {+driver-short+} 
to connect to your MongoDB cluster and run a query on the sample data.

Open the file named ``Program.cs`` in the base directory of your project. Copy the
following sample code into ``Program.cs`` 

.. literalinclude:: /includes/quick-start/Program.cs
   :language: csharp
   :dedent:

> The sample code runs a query against your sample dataset in MongoDB Atlas. Run the 
> sample code with the following command from your command line:

Reviewer 2: Suggestion:

1) I prefer using "this" over "the" when referring to an instance of something introduced just prior because it's more specific.
2) Sometimes there's a fine line between possessive "your" and non-possessive "the".
3) Reorder the final sentence so "command" is closest to the command itself which can be helpful for understanding its purpose.

```suggestion
This sample code runs a query against your sample dataset in MongoDB Atlas. Run it
from your command line using the following command:
```

Author: Changed "using" to "by using" as required by Style Guide, but adopted all other suggested changes

the {+driver-short+} to connect to your MongoDB cluster, run a query on the
sample data, and print out the result.

To learn more about connecting to Atlas with the {+driver-short+}, see
the :atlas:`Atlas driver connection </driver-connection>` guide

> and select **{+language+}** from the :guilabel`Select your language` dropdown.

Reviewer 2: Suggestion:

I think consistent usage of guilabel could help as suggested in a [previous comment](https://github.com/mongodb/docs-csharp/pull/74#discussion_r1097964746):

```suggestion
and select :guilabel:`C# / .NET` from the :guilabel`Select your language` dropdown.
```

Author: Thought I committed that change last time. Must have gotten lost when reconciling commits on GitHub and VSCode.

the {+driver-short+} to connect to your MongoDB cluster, run a query on the
sample data, and print out the result.

To learn more about connecting to Atlas with the {+driver-short+}, see
the :atlas:`Atlas driver connection </driver-connection>` guide

> and select **{+language+}** from the *Select your language* dropdown.

Reviewer 2: ```suggestion
and select **{+language+}** from the :guilabel`Select your language` dropdown.
```

Reviewer 2: I think the language should probably also use the guilabel per the style guide. I'm not sure how useful using a source constant is here since it should match the Atlas UI representation which could change.
